### PR TITLE
Export hidden pages in context

### PR DESCRIPTION
**Pull request by jml**
The `PageGenerator` was building hidden pages, but not making them
available in the context. This makes it difficult for other plugins to
operate on hidden pages.

This patch updates `PageGenerator` to export the hidden pages it finds
in the context as `hidden_pages`.

[Review on Reviewable](https://reviewable.io/reviews/getpelican/pelican/1760)

### CODE REVIEW

**Comment by ingwinlu** [comment]
maybe add the same behaviour for drafts as well

**Comment by jml** [comment]
Done. Thanks.

**Comment by ingwinlu** [comment]
you still need to update the docs https://github.com/getpelican/pelican/blob/master/docs/themes.rst

File: pelican/generators.py [hunk 1]
```diff
@@ -643,7 +643,7 @@ def generate_context(self):
         self.hidden_pages, self.hidden_translations = (
             process_translations(hidden_pages))
 
-        self._update_context(('pages', ))
+        self._update_context(('pages', 'hidden_pages'))
         self.context['PAGES'] = self.pages
```
Comment (ingwinlu) [inline pelican/generators.py]: should probably be removed; it is no where documented that pages is available as `pages` and `PAGES` + it is the (only?) variable I know of that actually is made available as UPPERCASE.

If suggest deprecating `PAGES`, issuing a warning if it is called, and update the default themes to use pages instead of PAGES.

@avaris @justinmayer thoughts?

Comment (avaris) [inline pelican/generators.py]: "issue a warning if it is called" will be tricky. Otherwise, I agree.

Comment (jml) [inline pelican/generators.py]: That sounds like a great thing to do in a separate PR.

Comment (justinmayer) [inline pelican/generators.py]: I agree that should be changed, but in a separate PR as @jml suggested.

**Comment by justinmayer** [comment]
Many thanks for the contribution, Jonathan. Now that the documentation has been added, would you be so kind as to squash the commits prior to merge?

**Comment by jml** [comment]
Thanks, am happy to help. Commits squashed.

**Comment by jml** [comment]
Anything else I need to do?

**Comment by justinmayer** [comment]
Thanks for the enhancement, Jonathan.
